11-Backbonejs/07-JavaScript-Objects-to-Backbone-Models.md

```diff
@@ -0,0 +1,230 @@
+
+Next you and your partner finish migrating the rest of the functions into the Backbone Model.
+
+Then run your tests with `npm test`
+
+### Check-in
+
+When you finish you can check your code [here](https://gist.github.com/CheezItMan/3034de6f979b27e36144bf14b1088018):  
+
+### Creating Instances of Player
+
+You can then create new player instances with:
+
+```javascript
+var player1 = new Player( {
+  name: 'Ms. Lovelace'
+});
+```
+
+### Making it Backbony!
+
+For this we need to understand the difference between Backbone attributes (which you get and set with the `get()` and `set()` functions).  Like we did above, we've added instance variables `plays` and `name` to instances of the model.  The alternative is to use Backbone attributes set in the `defaults` object or using the `set` function.  
+
+#### Why Use Backbone Attributes?
+
+For what we're doing so far, Backbone Attributes are a bit of overkill, but they do provide a few features which could be useful in... Tic-Tac-Toe!
+
+1.  Changing a Backbone Attribute triggers the `change` event on the model, which can be critical in rendering a model using a view.  
+1.  Backbone has a `fetch()` and a `save()` function which could potentially help you connect a model with a server's API to get and save data, assuming the attributes match attributes specified in the API.  Hmmm... maybe reading an API ahead of time is a good idea.  Don't worry, if they don't exactly match, there's a work-around.  
+
+You can convert plays and name to attributes here.
+
+```javascript
+// src/player.js
+const Player = Backbone.Model.extend({
+  defaults: {
+    plays: [],
+    name: "Player-?"
+  },
+  initialize: function(options) {
+    this.scrabble = new Scrabble();
+  },
+	...
+```
+
+People create a new player like this:
+
+```javascript
+var player1 = new Player({
+  name: 'Ada Lovelace'
+  });
+```
+
+
+
+
+## What!  All My Tests Broke!
+
+Yes, the tests were written using instance variables instead of using `get()` and `set()`, and yes **OMG** that's a lot to change.  
+
+Don't worry, you can find working tests [here](https://gist.github.com/CheezItMan/2d617fc903bf381c549b2859cc13c0bf).
+
+The primary changes in the JavaScript tests are:
+
+1.  Convert direct changes of an attribute to use the `set()` function.  `player1.name = "bob";` to `player1.set('name', 'bob');`
+1.  Convert reads of the attribute to use the `get()` function.
+1.  You will have to change the `beforeEach()` function to create the Player in a Backbone fashion
+
+
+## JS Objects vs Backbone Models
+
+Backbone Models are JavaScript Objects with a number of additional functions inherited from Backbone.Model.  
+
+When you use Backbone Models you need to remember:
+
+1.  You can decide how the model is set up with the `initialize()` function.  
+1.  Instance functions are passed in via an object in the 1st argument to the extend function.  
+1.  Static functions are passed in via an optional 2nd argument to extend.  
```

Comment:
Since we're not presenting them with any static functions, and most of the students will not have created their Tic-Tac-Toe project with any static functions, we might be able to leave this out.